purchase_date_planned_manual/models/purchase_order.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    def _compute_predicted_arrival_late(self):
+        """Colour the lines in red if the products are predicted to arrive
+        late."""
+        for line in self:
+            seller = line.product_id._select_seller(
+                partner_id=line.partner_id,
+                quantity=line.product_qty,
+                date=line.order_id.date_order and
+                line.order_id.date_order[:10],
+                uom_id=line.product_uom)
+            order_date = fields.Datetime.from_string(line.order_id.date_order)
+            po_lead = line.order_id.company_id.po_lead
+            delta = po_lead + seller.delay if seller else po_lead
+            date_expected = fields.Datetime.to_string(
+                order_date + relativedelta(days=delta))
+            line.predicted_arrival_late = (
+                date_expected > line.date_planned and line.order_id.state ==
+                'draft')
+
+    predicted_arrival_late = fields.Boolean(
```

Comment:
@mreficent Please put field definition on top and quote compute method

Reply:
why?

Reply:
No I'm not so mad :smile:  but because of https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models